x-pack/elastic-agent/pkg/agent/storage/sync_on_save_store.go

```diff
@@ -0,0 +1,81 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package storage
+
+import (
+	"io"
+	"os"
+	"runtime"
+
+	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/agent/errors"
+)
+
+// SyncOnSaveStore syncs paths after successful write.
+// we use this sync store to avoid issues seen on
+// low spec windows environment where agent is faster
+// than filesystem and what we read is different(stale)
+// than what we just wrote.
+type SyncOnSaveStore struct {
```

Comment:
Not sure we really need this type/functionality. The DiskStore did not fsync the file before closing it. Important files that are supposed to be valid even in case of crashes should always be synced before closing.